logbook-core/src/main/java/org/zalando/logbook/JsonHttpLogFormatter.java

```diff
@@ -101,10 +104,14 @@ private void addBody(final HttpMessage request, final ImmutableMap.Builder<Strin
         final String body = request.getBodyAsString();
 
         if (isJson(request.getContentType())) {
-            builder.put("body", body.isEmpty() ? JsonBody.EMPTY : new JsonBody(compactJson(body)));
-        } else {
-            addUnless(builder, "body", request.getBodyAsString(), String::isEmpty);
+            try {
+                builder.put("body", body.isEmpty() ? JsonBody.EMPTY : new JsonBody(compactJson(body)));
+                return;
```

Comment:
I tried this but then some tests were failing (those that are checking that for JSON an empty body is serialized as an empty string and for all other content-type the body is omitted). I'd go with omitting the body for an empty body, regardless of the content-type.